searchlib/src/vespa/searchlib/parsequery/simplequerystack.cpp

```diff
@@ -162,54 +163,58 @@ SimpleQueryStack::StackbufToString(const vespalib::stringref &theBuf)
     while (p < ep) {
         vespalib::string metaStr;
         rawtype = *p++;
-        type = search::ParseItem::GetType(rawtype);
-        if (search::ParseItem::GetFeature_Weight(rawtype)) {
+        type = ParseItem::GetType(rawtype);
+        if (ParseItem::GetFeature_Weight(rawtype)) {
             int64_t tmpLong(0);
             p += vespalib::compress::Integer::decompress(tmpLong, p);
             metaStr.append("(w:");
             metaStr.append(make_string("%ld", tmpLong));
             metaStr.append(")");
         }
-        if (search::ParseItem::getFeature_UniqueId(rawtype)) {
+        if (ParseItem::getFeature_UniqueId(rawtype)) {
             p += vespalib::compress::Integer::decompressPositive(tmp, p);
             metaStr.append("(u:");
             metaStr.append(make_string("%ld", tmp));
             metaStr.append(")");
         }
-        if (search::ParseItem::getFeature_Flags(rawtype)) {
+        if (ParseItem::getFeature_Flags(rawtype)) {
             flags = *p++;
             metaStr.append("(f:");
             metaStr.append(make_string("%d", flags));
             metaStr.append(")");
         }
-        if (search::ParseItem::GetCreator(flags) != search::ParseItem::CREA_ORIG) {
+        if (ParseItem::GetCreator(flags) != ParseItem::CREA_ORIG) {
             metaStr.append("(c:");
-            metaStr.append(make_string("%d", search::ParseItem::GetCreator(flags)));
+            metaStr.append(make_string("%d", ParseItem::GetCreator(flags)));
             metaStr.append(")");
         }
 
         metaStr.append('/');
         result.append(metaStr);
 
         switch (type) {
-        case search::ParseItem::ITEM_OR:
-        case search::ParseItem::ITEM_AND:
-        case search::ParseItem::ITEM_EQUIV:
-        case search::ParseItem::ITEM_NOT:
-        case search::ParseItem::ITEM_RANK:
-        case search::ParseItem::ITEM_ANY:
+        case ParseItem::ITEM_OR:
+        case ParseItem::ITEM_AND:
+        case ParseItem::ITEM_EQUIV:
+        case ParseItem::ITEM_NOT:
+        case ParseItem::ITEM_RANK:
+        case ParseItem::ITEM_ANY:
             p += vespalib::compress::Integer::decompressPositive(tmp, p);
             arity = tmp;
             result.append(make_string("%c/%d~", _G_ItemName[type], arity));
             break;
-        case search::ParseItem::ITEM_WEAK_AND:
-        case search::ParseItem::ITEM_NEAR:
-        case search::ParseItem::ITEM_ONEAR:
+        case ParseItem::ITEM_WEAK_AND:
+        case ParseItem::ITEM_SAME_ELEMENT:
```

Comment:
should move up to other items with only arity

Reply:
Code duplicated and for easier readability.